_posts/2022/2022-01-31-infrastructure-software-we-use-to-run-our-saas.md

```diff
@@ -0,0 +1,181 @@
+
+Added to our Heroku CI to parallelise it. The integration was very easy and it cut our builds to under 5 minutes. Most of that time is during setup: installing a browser for our system tests.
+
+### Performance, Errors & Uptime: [AppSignal](https://appsignal.com)
+
+Not as deep technically as some APM software, but at the right level for us and covers a bunch of infrastucture monitoring bases. Works nicely with Heroku. For instance the AppSignal PostgreSQL dashboard gives better insight than the (frankly poor) Heroku version.
+
+And they send out [stroopwafels](https://www.appsignal.com/waffles).
+
+
+### Application Security: [Sqreen](https://sqreen.io)
+
+One piece of software that I'm especially glad to have found. It constant reviews and protects our applications from both deliberate attacks from troublemakers and our own inadvertanty security issues.
+
+Purchased by Datadog, I guess they’ll wind that into their main product at some point.
+
+
+### Logging: [LogDNA](https://www.logdna.com)
+
+A nice UI and reasonable pricing for our volume.
+
+
+### Static Site Hosting: [Netlify](https://netlify.com)
+
+Our static sites are deployed here, we also make some use of a couple of very simple JavaScript “serverless” functions that are called from our applications.
+
+It has a similarly straightforward deployment approach to Heroku, which is one of the main reasons we use it.
+
+
+### CDN: [Fastly](https://fastly.com)
+
+Solid CDN. The main players are all much of a muchness in this arena. We're only (currently) using it for our Rails assets.
+
+
+### Payment Processing: [Stripe](https://stripe.com)
+
+Why would we use anything else to deal with our credit card transactions? Arguably without it, the business would have struggled to get off the ground.
+
+I speak as a man who previously implemented online credit card processing directly with banks. It was horrible.
+
+
+### Code Collaboration: [Github](https://github.com)
+
+Where our code is stored, reviewed and merged. Pretty standard for a modern development team.
+
+We have a couple of Github Actions that we use for linting our code and compressing our images before merging.
+
+We also use the [dependabot](https://github.com/dependabot) functionality Github absorbed.
+
+
+### Image CDN: [Imgix](https://imgix.com)
+
+There’s a bunch of products that perform this task but their API is the most pleasant of any we’ve seen and their pricing is fair. Good and engaged support.
+
+
+### Video CDN: [Mux](https://mux.com)
+
+Our video CDN of choice. Lovely product, good on site player and very nice API.
+
+
+### Video: [Wistia](https://wistia.com)
+
+We serve some of our video marketing assets through here, we also use YouTube.
+
+
+### Cloud Storage: [AWS S3](https://aws.amazon.com)
+
+We use Amazon’s storage because it’s the industry default, we should probably look at alternatives. AWS’s dashboard terrifies me every time I open it.
```

Comment:
Again, full stop after default.

And if you're going to say that that you should look into alternatives, might be worth stating why?

Reply:
```suggestion
We use Amazon’s storage because it’s the industry default. AWS’s dashboard terrifies me every time I open it.
```